src/main/java/com/qunite/api/security/JwtService.java

```diff
@@ -3,32 +3,78 @@
 import com.auth0.jwt.JWT;
 import com.auth0.jwt.algorithms.Algorithm;
 import com.auth0.jwt.interfaces.DecodedJWT;
+import com.qunite.api.data.UserRepository;
+import com.qunite.api.domain.Tokens;
 import com.qunite.api.domain.User;
+import com.qunite.api.service.TokenService;
+import com.qunite.api.web.dto.auth.AuthenticationResponse;
 import java.time.Instant;
+import java.time.temporal.ChronoUnit;
 import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
-@Component
+@Service
+@RequiredArgsConstructor
 public class JwtService {
   @Value("${jwt.secret}")
   private String secret;
 
   @Value("${spring.application.name}")
   private String issuer;
 
-  public String createJwtToken(User user) {
+  private final TokenService tokenService;
+
+  private final UserRepository userRepository;
+
+  @Transactional
+  public AuthenticationResponse createJwtTokens(User user) {
+    String accessToken = generateJwtToken(
+        user, 30, ChronoUnit.MINUTES, TokenType.ACCESS_TOKEN);
+    String refreshToken = generateJwtToken(
+        user, 7, ChronoUnit.DAYS, TokenType.REFRESH_TOKEN);
+
+    Tokens tokens = new Tokens();
+    tokens.setAccessToken(accessToken);
+    tokens.setRefreshToken(refreshToken);
+    tokens.setOwner(user);
+    tokenService.createOne(tokens);
+
+    return new AuthenticationResponse(accessToken, refreshToken, "JWT", "HS256", 30 * 60);
```

Comment:
1800

Reply:
It`s actually better to take expiration time to some env variable, but it requires updating docker compose e.t.c, so besides it`s better to read, is`s better to store the same value in one dimension(minutes) rather than hardcodly convert it to seconds.